### ISSUE TITLE
Animation lag

### ISSUE THREAD

**Issue by paurbo** [tomaszpoliszuk/MagSafeController#6]
I’m on an iPhone XR, iOS 13.5. The animation lags at first but ends smooth. [Here is a video of it.](https://imgur.com/a/Y8fsNTk)

**Comment by tomaszpoliszuk** [comment]
If by lag you mean that moment when splash part (white background circle) seems to kind of pause for a moment and then proceed to expand and eventually fade off then it is byproduct of using [Ease In Ease Out](https://github.com/tomaszpoliszuk/MagSafeController/blob/cdc9cf7eaaa4f05bdbc688c6ac6d88fddb2d8a46/MagSafeController.xm#L53) in [splash element](https://github.com/tomaszpoliszuk/MagSafeController/blob/cdc9cf7eaaa4f05bdbc688c6ac6d88fddb2d8a46/MagSafeController.xm#L183) and i've decide to leave it for a moment like it is now (IDK it by accident automatically brings water splash to my memory) but it will most likely cease to exist in next phase of development when i will upgrade animation to better reflect one made by Apple in iOS 14.1

**Comment by paurbo** [comment]
Yeah it seems to pause twice I think. I was just comparing it to the video on the tweak page in which it seems quite a lot more smooth. I guess that one is the native one. Apart from the pauses they are very similar - great job!

**Comment by tomaszpoliszuk** [comment]
No - that video is from custom animation i"ve made - just to be 100% sure - could you try to disable other tweaks and test how this animation looks then (but I'm 95% sure this is correct - at least after comparing code and video - behavior).

**Comment by IORose** [comment]
> I’m on an iPhone XR, iOS 13.5. The animation lags at first but ends smooth. [Here is a video of it.](https://imgur.com/a/Y8fsNTk)

I have the same issue:
iPhone 11
iOS 13.5
unc0ver 5.3.1

https://youtu.be/GPtRlFuCD3E

**Comment by tomaszpoliszuk** [comment]
It was intentional at time of writing this animation but yeah - it looks different than original MagSafe animation and seems like it's lagging.
I'm in process of rewriting whole custom part (ideally to have identical animation as it's in iOS 14.1 and up) so changes are coming but it's hard to say when it will happen.

I'll keep this issue open until i publish updated version of this animation.
